gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionCompletionContributor.java

```diff
@@ -47,45 +55,114 @@ public class VersionCompletionContributor extends CompletionContributor {
-                        return version.isLatest()
-                                ? PrioritizedLookupElement.withPriority(
-                                        LookupElementBuilder.create(version)
-                                                .withTypeText("Latest", true)
-                                                .withLookupString("latest"),
-                                        Double.MAX_VALUE)
-                                : PrioritizedLookupElement.withPriority(LookupElementBuilder.create(version), priority);
+                        List<GroupAndDep> groupAndDeps = collectGroupAndDeps(project, group, dependencyName);
+                        addToResults(sortedResultSet, groupAndDeps);
                     }
                 });
     }
 
+    private DependencyInfo getDependencyInfo(CompletionParameters parameters) {
+        VersionPropsDependencyVersion versionElement = ReadAction.compute(
+                () -> (VersionPropsDependencyVersion) parameters.getPosition().getParent());
+
+        VersionPropsProperty property = ReadAction.compute(() -> findParentProperty(versionElement));
+
+        DependencyGroup group =
+                DependencyGroup.fromString(property.getDependencyGroup().getText());
+        DependencyName dependencyName =
+                DependencyName.of(property.getDependencyName().getText());
+
+        return new DependencyInfo(group, dependencyName);
+    }
+
+    private void addLoadingElement(CompletionResultSet sortedResultSet) {
+        LookupElement loadingElement = PrioritizedLookupElement.withPriority(
+                LookupElementBuilder.create("Loading Versions...").withInsertHandler((elementContext, item) -> {
+                    // Prevent insertion
+                    elementContext
+                            .getDocument()
+                            .deleteString(elementContext.getStartOffset(), elementContext.getTailOffset());
+                }),
+                Double.MIN_VALUE);
+        sortedResultSet.addElement(loadingElement);
+    }
+
+    private void handleDependencyWithoutWildcard(
+            CompletionResultSet sortedResultSet,
+            Project project,
+            DependencyGroup group,
+            DependencyName dependencyName) {
+        RepositoryLoader.loadRepositories(project)
+                .forEach(url -> addToResults(sortedResultSet, List.of(new GroupAndDep(group, dependencyName, url))));
+    }
+
+    private List<GroupAndDep> collectGroupAndDeps(
+            Project project, DependencyGroup group, DependencyName dependencyName) {
+        String dependencyNamePrefix = dependencyName.name().replace("*", "");
+        return StreamEx.of(RepositoryLoader.loadRepositories(project))
+                .flatMap(url -> StreamEx.of(GROUP_PART_OR_PACKAGE_NAME_EXPLORER.getGroupPartOrPackageName(
+                                group, url, RefreshUtil.refreshOnceSupplier()::get))
```

Comment:
I don't think there's any point in scheduling a refresh here as we haven't loaded any versions into the completions yet. Similarly for the package completions, we probably don't need to refresh as we don't care so much about the ordering like we do with versions (versions we always want latest to be in the right place, packages people are typing partial parts of the name which is more "normal").

Reply:
I'd probably just make this block as it did before, maybe loading everything in parallel. Especially as most of the time it's just parsing a whole load of links on a single html page?

Reply:
Maybe it would make more sense for a `com.*:*` pattern when there are multiple things to load.

Reply:
So I have converted it to being synchronous again and added the progress indicator logic back in for the `GroupPartOrPackageNameExplorer` to prevent it blocking the UI - I think a future PR can worry about `*:*`